ristretto/ristretto.go

```diff
@@ -0,0 +1,93 @@
+package ristretto
+
+import (
+	"time"
+
+	"github.com/dgraph-io/ristretto"
+)
+
+// Storage interface that is implemented by storage providers.
+type Storage struct {
+	cache       *ristretto.Cache
+	defaultCost int64
+}
+
+// New creates a new storage.
+func New(config ...Config) *Storage {
+	cfg := configDefault(config...)
+	cache, err := ristretto.NewCache(&ristretto.Config{
+		NumCounters: cfg.NumCounters,
+		MaxCost:     cfg.MaxCost,
+		BufferItems: cfg.BufferItems,
+	})
+
+	if err != nil {
+		return nil
+	}
+
+	store := &Storage{
+		cache:       cache,
+		defaultCost: cfg.DefaultCost,
+	}
+
+	return store
+}
+
+// Get gets the value for the given key.
+// `nil, nil` is returned when the key does not exist
+func (s *Storage) Get(key string) ([]byte, error) {
+	if len(key) <= 0 {
+		return nil, nil
+	}
+
+	item, found := s.cache.Get(key)
+	if !found {
+		return nil, nil
+	}
+
+	buf, asserted := item.([]byte)
+	if !asserted {
+		return nil, nil
+	}
+
+	return buf, nil
+}
+
+// Set stores the given value for the given key along
+// with an expiration value, 0 means no expiration.
+// Empty key or value will be ignored without an error.
+func (s *Storage) Set(key string, val []byte, exp time.Duration) error {
+	if len(key) <= 0 || len(val) <= 0 {
+		return nil
+	}
+	saved := s.cache.SetWithTTL(key, val, s.defaultCost, exp)
+	if !saved {
+		return nil
+	}
+	// wait for value to pass through buffers
+	time.Sleep(10 * time.Millisecond)
```

Comment:
Can you explain a bit more what this is for?

Reply:
Actually, I don't really know what is this for. It is stated in the documentation of ristretto here is the example [link](https://github.com/dgraph-io/ristretto#example). But I believe it is safe to delete this line as it is not important